lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -685,12 +690,20 @@ WasmBinaryReader::FunctionBodyHeader()
     TRACE_WASM_DECODER(L"Function body header: i32 = %u, i64 = %u, f32 = %u, f64 = %u", i32Count, i64Count, f32Count, f64Count);
 }
 
+void
+WasmBinaryReader::DataSegment()
+{
+    UINT len = 0;
+    WasmDataSegment *dseg = Anew(&m_alloc, WasmDataSegment, &m_alloc, LEB128(len), LEB128(len), m_pc);
```

Comment:
don't you need to do `m_pc += size` (and ideally validate against reading beyond end)?